cfme/cloud/keypairs.py

```diff
@@ -119,24 +119,67 @@ def is_displayed(self):
     form = View.nested(KeyPairAddForm)
 
 
-class KeyPair(Taggable, SummaryMixin, Navigatable):
+class KeyPairCollection(Navigatable):
+    """ Collection object for the :py:class: `cfme.cloud.KeyPair`. """
+
+    def instantiate(self, name, provider, public_key):
+        return KeyPair(name, provider, public_key=public_key or "", collection=self)
+
+    def create(self, name, provider, public_key=None, cancel=False):
+        """Create new keyPair.
+
+        Args:
+            name (str): name of the KeyPair
+            public_key (str): RSA Key if present
+            provider (str): Cloud Provider
+            cancel (boolean): Cancel Keypair creation
+        """
+
+        view = navigate_to(self, 'Add')
+        changed = view.form.fill({'name': name,
+                                  'public_key': public_key,
+                                  'provider': provider.name
+                                  })
+        if cancel and not changed:
+            view.form.cancel.click()
+            flash_message = 'Add of new Key Pair was cancelled by the user'
+        else:
+            view.form.add.click()
+            flash_message = VersionPick({
+                Version.lowest(): 'Creating Key Pair {}'.format(name),
+                '5.8': 'Key Pair "{}" created'.format(name)}).pick(self.appliance.version)
+
+        # add/cancel should redirect, new view
+        view = self.create_view(KeyPairAllView)
+        # TODO BZ 1444520 causing ridiculous redirection times after submitting the form
+        wait_for(lambda: view.is_displayed, num_sec=120, delay=3,
+                 fail_func=view.flush_widget_cache(), handle_exception=True)
```

Comment:
@sbulage you put `view.flush_widget_cache()` instead of `view.flush_widget_cache`